pkg/parser/misc.go

```diff
@@ -29,6 +29,17 @@ func isIdentExtend(ch byte) bool {
 	return ch >= 0x80
 }
 
+// See https://dev.mysql.com/doc/refman/5.7/en/identifiers.html
+func isInCorrectIdentifierName(name string) bool {
```

Comment:
This method is same as `IsInCorrectIdentifierName` in util package.
https://github.com/yoshikipom/tidb/blob/b2debdab385e36c0f8953ed7244473bbed1f3286/pkg/util/util.go#L290

But we can't use util package from parser package because they are not in same go project. Although it's possible to import `util` from `parser`, I chose copy code to avoid cyclic dependency. The main go project including `util` imports `parser`.
https://github.com/pingcap/tidb/blob/master/go.mod#L88